pvm/README.md

```diff
@@ -0,0 +1,33 @@
+# PVM Test Vectors, version 0.1
+
+## How to use this
+
+The [`programs`](./programs) directory contains `.json` files, each containing a single test.
+
+Here's an example of such a test:
+
+```
+{
+  "name": "inst_add",
+  "initial_regs": [0, 0, 0, 0, 0, 0, 0, 1, 2, 0, 0, 0, 0],
+  "code": [0, 0, 3, 8, 121, 8, 249],
+  "expected_status": "trap",
+  "expected_regs": [0, 0, 0, 0, 0, 0, 0, 1, 2, 3, 0, 0, 0]
+}
+```
+
+* `name` -- a unique identifier for the test
+* `initial_regs` -- the initial value of each of the 13 registers; these need to be set *before* the test program is executed
+* `code` -- the code blob of the program to be executed as part of the test
+* `expected_status` -- the way the program is supposed to end; currently it can be one of the following:
+   - `"trap"` -- the execution ended with a trap (the `trap` instruction was executed, the execution went "out of bounds", an invalid jump was made, or an invalid instruction was executed)
+   - `"halt"` -- the execution finished gracefully (a dynamic jump to address `0xffff0000` was made)
+* `expected_regs` -- the expected values of each of the 13 registers *after* the test program is executed
```

Comment:
Would be nice if these could be done to the level of the Safrole test vectors - i.e. referencing the relevant symbols and equations in the GP.

Specifically, it looks like these test vectors are for (207) **Psi**, but omit the arguments and results for gas, memory and initial instruction counter.

Reply:
> Would be nice if these could be done to the level of the Safrole test vectors - i.e. referencing the relevant symbols and equations in the GP and providing a good ASN definition.

Sure, I can do that.

> Specifically, it looks like these test vectors are for (207) Psi, but omit the arguments and results for gas, memory and initial instruction counter.

Yes. The omissions were intentional, since no tests use them right now. This is only temporary of course.